bingo.js

```diff
@@ -598,13 +670,142 @@ function fillVersionSelection()
 	});
 }
 
-function setSquareColor(square, colorClass)
+function setSquareColor(square, colorClass, isTriggeredByUser = false)
 {
 	ALL_COLOURS.forEach(c => square.removeClass(c));
 	square.addClass(colorClass);
+	if (isTriggeredByUser) {
+		updateCurrentSheetProgress();
+	}
 }
 
-function copySeedToClipboard(id, event)
+function loadBingoProgress(sheetToLoad)
+{
+	/* 
+	 * Expected input for sheetToLoad is [1,3,2,5,..]
+	 * where index of array is numeric value of square id
+	 * and cell value is index of ALL_COLOURS 
+	 */
+	if (typeof sheetToLoad !== 'undefined' && sheetToLoad.length) {
+		forEachSquare((i, square) => {
+			$.each(ALL_COLOURS, function(index, colorClass){
+				square.removeClass(colorClass);
+			});
+			square.addClass(ALL_COLOURS[sheetToLoad[i]]);
+		});
+	}
+}
+
+var sheetProgressDebounce = null;
+function updateCurrentSheetProgress()
+{
+	if(sheetProgressDebounce) clearTimeout(sheetProgressDebounce);
+	sheetProgressDebounce = setTimeout(function() {
+			updateCurrentSheetProgressDebounced()
+	}, 100);
+}
+
+function updateCurrentSheetProgressDebounced()
+{
+	if (! isLocalStorageIsAvailable()) {
+		return;
+	}
+
+	processSheetAndStoreAs(getKeyForCurrentProgress());
+}
+
+function processSheetAndStoreAs(localStorageKey)
```

Comment:
Function `processSheetAndStoreAs` is called once. The parameter `localStorageKey` is always the result of `getKeyForCurrentProgress()`. I suggest converting the parameter into a local variable and renaming the function to `storeProgress` for symmetry with the `loadProgress*` functions.